idaes/core/util/math.py

```diff
@@ -191,3 +191,16 @@ def safe_log(a, eps=1e-4):
         approximately log(max(a, eps))
     """
     return log(smooth_max(a, eps, eps=eps))
+
+
+def smooth_heaviside(x, k):
+    """
+    Provides a smooth, continuous approximation of a discontinuous function
+    Args:
+        x : Independent variable; as x increases, the function increases from 0 to 1
+        k : Smoothing parameter representing the slope of the discontinuity
+    Returns:
+        function : Continuous approximation of a discontinuous function
+    """
+    function = 1 / (1 + exp(-2 * k * x))
```

Comment:
Out of curiosity, why `-2 * k * x` instead of just `-k * x`? I thought it might be to get the derivative to equal 1 at `x=0`, but that requires `-4*k*x` instead.

(The answer might be that you got it from me during our original discussions, in which case I have no idea why I chose it.)

Reply:
Yea, this was pulled from our original discussions. That being said, I can (and probably should) still remove it so that we're not propagating numbers that we can't explain.